src/catalog/catalog_entry/sequence_catalog_entry.cpp

```diff
@@ -0,0 +1,123 @@
+#include "catalog/catalog_entry/sequence_catalog_entry.h"
+
+#include "binder/ddl/bound_create_sequence_info.h"
+#include "common/exception/catalog.h"
+#include "common/exception/overflow.h"
+#include "function/arithmetic/add.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace catalog {
+
+SequenceData SequenceCatalogEntry::getSequenceData() {
+    std::lock_guard<std::mutex> lck(mtx);
+    return sequenceData;
+}
+
+int64_t SequenceCatalogEntry::currVal() {
+    std::lock_guard<std::mutex> lck(mtx);
+    int64_t result;
+    if (sequenceData.usageCount == 0u) {
```

Comment:
Correct me if I'm wrong here.
We seem to only care about `usageCount` being 0 or not. then why not use a boolean flag instead? Isn't that more intuitive?

Reply:
Update: need usageCount to consider the case where two transactions make changes to the same sequence.